localstack/aws/handlers/cors.py

```diff
@@ -152,8 +152,9 @@ def should_enforce_self_managed_service(context: RequestContext) -> bool:
 
     if not config.DISABLE_CUSTOM_CORS_APIGATEWAY:
         # we don't check for service_name == "apigw" here because ``.execute-api.`` can be either apigw v1 or v2
-        is_user_request = (
-            ".execute-api." in context.request.host or PATH_USER_REQUEST in context.request.path
+        path = context.request.path
+        is_user_request = ".execute-api." in context.request.host or (
+            path.startswith("/restapis") and f"/{PATH_USER_REQUEST}" in context.request.path
```

Comment:
This check seems good to me, we could just add the trailing slash to `restapis` here as well I guess:
```suggestion
            path.startswith("/restapis/") and f"/{PATH_USER_REQUEST}" in context.request.path
```